apps/carbon/app/routes/x+/quote+/$id.convert-to-order.tsx

```diff
@@ -18,110 +19,35 @@ export async function action(args: ActionFunctionArgs) {
   const { request, params } = args;
   assertIsPost(request);
 
-  const { client, userId } = await requirePermissions(request, {
+  const { userId } = await requirePermissions(request, {
     create: "sales",
     role: "employee",
   });
 
   const { id } = params;
-  if (!id) throw new Error("Could not find id");
-
-  let newSalesOrderId: string;
-
-  const updateQuoteStatus = await convertQuoteToOrder(client, id, userId);
-  if (updateQuoteStatus.error) {
-    throw redirect(
-      path.to.quote(id),
-      await flash(
-        request,
-        error(
-          updateQuoteStatus.error,
-          "Failed to update quote status to Ordered"
-        )
-      )
-    );
-  }
-
-  const quote = await getQuote(client, id);
-  if (quote.error) {
-    throw redirect(
-      path.to.quote(id),
-      await flash(request, error(quote.error, "Failed to get quote"))
-    );
-  }
+  if (!id) throw new Error("Could not find quote id");
 
+  let salesOrderId = "";
   try {
-    const nextSequence = await getNextSequence(client, "salesOrder", userId);
-    if (nextSequence.error) {
-      throw redirect(
-        path.to.newSalesOrder,
-        await flash(
-          request,
-          error(nextSequence.error, "Failed to get next sequence")
-        )
-      );
-    }
-    const createSalesOrder = await upsertSalesOrder(client, {
-      salesOrderId: nextSequence.data,
-      createdBy: userId,
-      quoteId: quote.data.id,
-      customerId: quote.data.customerId,
-      orderDate: new Date().toISOString(),
-    });
-
-    if (createSalesOrder.error || !createSalesOrder.data?.[0]) {
-      // TODO: this should be done as a transaction
-      await rollbackNextSequence(client, "salesOrder", userId);
+    const result = await createSalesOrderFromQuote(id, userId);
+    if (result.error || !result?.data) {
       throw redirect(
-        path.to.quotes,
+        request.headers.get("Referer") ?? path.to.quotes,
```

Comment:
There's a util method for this:

```suggestion
        requestReferrer(request) ?? path.to.quote,
```

Reply:
addressed and made it `path.to.quote(id)`